### DISCUSSION TITLE
Love the project but some questions about sweeps

### DISCUSSION THREAD

**Discussion by vinitjogani** [sgenoud/replicad#170] [How To]
I love how simple the API for this project is -- very intuitive compared to some of the others I have tried. Everything just works as you'd expect. Kudos!

That said, I am looking to do some advanced sweeps. In particular, I want to do a 2-rail sweep where there is an inner and outer wire, and a profile which has to be resized so that it spans the cross section between the wires.

Now I have some custom code that sort of achieves this in cadquery by just computing a bunch of cross sections at different intervals and lofting them through. However, to implement that in Replicad, I need a `getPositionAt(...)` function on a wire where I can pass in a 0-1 value and use that to sub-divide the wire and get positions at different thresholds. Is there something similar in the API that I can use? 

Also, are there any features for bending, shearing, tapering, etc. of models?

**Comment by sgenoud** [comment]
I have added a `wire.pointAt(index)` method to the `Wire`. It [should work once your studio has been updated](https://studio.replicad.xyz/workbench#code=UEsDBAoAAAAIACFGUllYU2lwLAEAAPYBAAAHAAAAY29kZS5qc01Qy27CMBC8%252BytWOdkoMgT6oEJURb0iFamXVqgHkyzFIrEj2xGtovx71yRQDh7t7szOrDwejeAl%252FNZY4B5aiJXdg65q6wJPHNalzlWRCOjg0qz1DkZjxsaXVWhvqC5yuTU%252BkF3h1CmFSh3xvT6gQ3JZXn0WjOFPzIFeTheopgwb5VTlSdd2%252F4qBg31j8qCtIU9tuICWwbDtjxjyA63FTC5oDiAPa22QzyZDmzc%252B2OrVOoOOz4fhrim%252FceVynk1TyDJ6A1FY2h3q3v3NbEpFw%252BTjMxGLa3RttQnx4u0khYnMIkwjzCLcRbiP8BDhMcI8whNFfclK1ZzXApbP56Q%252BSJ60Q8%252BFPFuvAgmIPUc6DI0zsO2FKUjZi%252FyN082P80zI4JTxpQpIrPhasI79AVBLAQIUAAoAAAAIACFGUllYU2lwLAEAAPYBAAAHAAAAAAAAAAAAAAAAAAAAAABjb2RlLmpzUEsFBgAAAAABAAEANQAAAFEBAAAAAA%253D%253D).

<img width="356" alt="Capture d’écran 2024-10-18 à 10 28 09" src="https://github.com/user-attachments/assets/6a2d1d74-3213-472f-9cd3-afa9e0970089">

Note that the mapping of the [0, 1] interval on the wire is not necessarily equally spaced (as you can see in the image).

I would be curious to see your 2-rail sweep - and potentially add it as a recipe in the doc.

**Comment by sgenoud** [comment]
> Also, are there any features for bending, shearing, tapering, etc. of models?

No, I do not really use these. Do you have some example of a use case (and potentially an API you have seen and like?)
